docs/book/src/migration/guide.md

```diff
@@ -0,0 +1,192 @@
+# Migration Guide
+
+This document describes how to migrate a project from kubebuilder v1 to v2.
+
+We suggest you read [this doc](./v1vsv2.md) that compares the notable differences between v1 and v2.
+
+The recommended way of migration is to create a new v2 project and copy the business logic code.
+
+Please ensure you have followed the [installation guide](../quick-start.md#installation) to install the required
+components.
+
+We have an [example v1 project](https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/migration/testdata/gopath/project). It should look like the
```

Comment:
move these into reference-style links to make the source more readable:

```markdown

We have an [example v1 project][v1-project]

[v1-project]: https://github.com/...
````

Reply:
Lang: We've been writing our tutorials in a more conversational tone, so this should probably read like

> Let's take the [example v1 project][v1-project] and migrate it to KubeBuilder v2.
> At the end, we should have something that looks like the [example v2 project][v2-project].